### ISSUE TITLE
cpp-ethereum failed to build on OS-X 10.9.5

### ISSUE THREAD

**Issue by benWoz** [ethereum/homebrew-ethereum#58]
I'm trying to install cpp-ethereum. When I do a "$ brew install cpp-ethereum" it fails-here's the current log file: 
https://gist.github.com/9c7ff96909ade78d4974

Previously, it failed like this:
====
[  4%] Built target scrypt
clang: error: unknown argument: '-fstack-protector-strong'
make[2]: *** [libethereum/libethash/CMakeFiles/ethash.dir/sha3_cryptopp.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [libethereum/libethash/CMakeFiles/ethash.dir/all] Error 2
make: *** [all] Error 2
====
I created a gist of this earlier logs:
https://gist.github.com/15c85696e931b9eafa1f

I tried these things to fix it:
$ export ARCHFLAGS="-Wno-error=unused-command-line-argument-hard-error-in-future"
$ brew install llvm --HEAD --with-clang

Thx, Ben

**Comment by bobsummerwill** [comment]
Thanks for the report.   Will try to have a look today.

**Comment by benWoz** [comment]
Sound great Bob! I appreciate your help. I really want to get my first Dapp
going! This seems like such a edgy space: the future, you know? If I can't
get the C++ library installed, maybe I'll give the Python version a shot,
but I hear C++ is the way to go for devs with Ethereum.

Thx, Ben

**Comment by bobsummerwill** [comment]
No problem.

Please do also check out geth, the go client (https://github.com/ethereum/go-ethereum/ | http://ethereum.github.io/go-ethereum/).   That is by far the most popular client, and might be your easiest path to a working setup.

Also, you can get started with Solidity development without even installing a client, in the browser:
https://ethereum.github.io/solidity/

And have you seen these?
https://www.reddit.com/r/ethereum/comments/3vxvlx/starter_guide_almost_all_the_links_youll_need_to/

**Comment by bobsummerwill** [comment]
Oh dear ... now I see the issue, and I'm afraid you might be out of luck.

The build error is because you have a very old Clang, because you have an old OS X (Mavericks), which we don't support.    Are you able to upgrade your OS on the hardware which you have?   If so, I would recommend you do so anyway, irrespective of this particular issue.

Sorry :-(    Best wishes!

**Comment by benWoz** [comment]
There we go! I have updated and it appears to have installed cpp-ethereum correctly. 

Thanks for digging on this for me.

What did I need to look for to figure this one out myself?

Thx, Ben

**Comment by bobsummerwill** [comment]
So this page (https://github.com/ethereum/webthree-umbrella/wiki/Building-on-OS-X) said "for best results use El Capitan, and it might work for older versions.

I've just updated it to be more specific, and to say we only support Yosemite and El Capitan and older versions are known broken and won't be fixed.

At the time I rewrote the original page a month or two back, I did see references to it working on Mavericks (or at least running on Mavericks even if it wasn't buildable).    Given your experience, and that of others, I think it's better just to chop that off entirely.

Thanks for unintentionally guinea-pigging the instructions!
